state-chain/pallets/cf-elections/src/lib.rs

```diff
@@ -2127,4 +2132,63 @@ pub mod pallet {
 			}
 		}
 	}
+
+	#[cfg(any(test, feature = "try-runtime"))]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Ensure the correctness of the state of this pallet.
+		///
+		/// These invariants should be valid before or after each state transition of this pallet.
+		///
+		/// # Invariants
+		/// * All keys in `ElectionProperties` and `ElectionState` should match
+		/// * All keys in `SharedData` should have an entry in `SharedDataReferenceCount`
+		/// * `ElectionConsensusHistoryUpToDate` has a corresponding entry in `ElectionProperties`
+		/// * `BitmapComponents` have a corresponding entry in `ElectionProperties`
+		/// * `IndividualComponents` have a corresponding entry in `ElectionProperties`
+		pub fn do_try_state() -> Result<(), DispatchError> {
+			let properties_keys = ElectionProperties::<T, I>::iter_keys()
+				.map(|id| *id.unique_monotonic())
+				.collect::<BTreeSet<_>>();
+
+			if properties_keys != ElectionState::<T, I>::iter_keys().collect::<BTreeSet<_>>() {
+				Err(DispatchError::Other(
+					"All keys in ElectionProperties and ElectionState should match",
+				))?;
+			}
```

Comment:
```suggestion
			if properties_keys != ElectionState::<T, I>::iter_keys().collect::<BTreeSet<_>>() {
				Err(DispatchError::Other(
					"All keys in ElectionProperties and ElectionState should match",
				))?;
			}
```

The substrate-y way to do this is to use ensure!() - is also nicer because we don't need to wrap it in Err()

Reply:
same for all of these conditions